### PR TITLE
Remove unused/unneeded list vars and logic for packages and TPLs (#299, #63)

### PR DESCRIPTION
**Pull request by bartlettroscoe**
## Description

This is a cleanup PR to allow the next step in working towards #63 (and removing some stuff not needed after the completion of #299).

See the individual commit messages for what was changed and why.

## Instructions for Reviewers

* Please review the commits individually (there are no intermediate commits with changes not reflected in the final diff).

### CODE REVIEW

**Comment by bartlettroscoe** [comment]
Hello @KyleFromKitware, can you please give this PR a review when you get some time?

I will build my next branch off of this branch to continue work in the meantime.

File: tribits/core/package_arch/TribitsSetUpEnabledOnlyDependencies.cmake [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+      tribits_package_set_full_enabled_dep_packages(${tribitsPackage})
+      if (${PROJECT_NAME}_DUMP_PACKAGE_DEPENDENCIES)
+        set(PRINTED_VAR FALSE)
+        print_nonempty_var_with_spaces(${tribitsPackage}_FULL_ENABLED_DEP_PACKAGES
+          PRINTED_VAR)
+        if (NOT PRINTED_VAR)
+          message("-- ${tribitsPackage}: No library dependencies!")
+        endif()
+      endif()
+      if ("${lastExportTribitsPackage}" STREQUAL ${tribitsPackage})
```
Comment (KyleFromKitware) [inline tribits/core/package_arch/TribitsSetUpEnabledOnlyDependencies.cmake:130]: Is `${tribitsPackage}` supposed to be quoted?

File: tribits/core/package_arch/TribitsSetUpEnabledOnlyDependencies.cmake [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+
+# Function that sets up the full package dependencies for the given internal
+# enabled package ``${PACKAGE_NAME}``, including all of its indirect upstream
+# internal package dependencies.
+#
+function(tribits_package_set_full_enabled_dep_packages  PACKAGE_NAME)
+
+  set(PACKAGE_FULL_DEPS_LIST "")
+
+  foreach(DEP_PKG ${${PACKAGE_NAME}_LIB_REQUIRED_DEP_PACKAGES})
```
Comment (KyleFromKitware) [inline tribits/core/package_arch/TribitsSetUpEnabledOnlyDependencies.cmake:148]: `IN LISTS`

File: tribits/core/package_arch/TribitsSetUpEnabledOnlyDependencies.cmake [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+    if (${PROJECT_NAME}_ENABLE_${DEP_PKG})
+      list(APPEND  PACKAGE_FULL_DEPS_LIST  ${DEP_PKG})
+    endif()
+    # NOTE: This if() should not be needed but this is a safeguard
+  endforeach()
+
+  foreach(DEP_PKG ${${PACKAGE_NAME}_LIB_OPTIONAL_DEP_PACKAGES})
```
Comment (KyleFromKitware) [inline tribits/core/package_arch/TribitsSetUpEnabledOnlyDependencies.cmake:155]: `IN LISTS`

File: tribits/core/package_arch/TribitsSetUpEnabledOnlyDependencies.cmake [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+    if (${PACKAGE_NAME}_ENABLE_${DEP_PKG})
+      list(APPEND  PACKAGE_FULL_DEPS_LIST  ${DEP_PKG})
+    endif()
+  endforeach()
+
+  if(PACKAGE_FULL_DEPS_LIST)
+    list(REMOVE_DUPLICATES  PACKAGE_FULL_DEPS_LIST)
+
+    foreach(DEP_PACKAGE  ${PACKAGE_FULL_DEPS_LIST})
```
Comment (KyleFromKitware) [inline tribits/core/package_arch/TribitsSetUpEnabledOnlyDependencies.cmake:164]: `IN LISTS`

File: tribits/core/package_arch/TribitsSetUpEnabledOnlyDependencies.cmake [hunk 5]
```diff
@@ -0,0 +1,205 @@
...
+      list(APPEND PACKAGE_FULL_DEPS_LIST  ${${DEP_PACKAGE}_FULL_ENABLED_DEP_PACKAGES})
+    endforeach()
+
+    list(REMOVE_DUPLICATES PACKAGE_FULL_DEPS_LIST)
+  endif()
+
+  set(ORDERED_PACKAGE_FULL_DEPS_LIST "")
+
+  foreach(DEP_PACKAGE  ${PACKAGE_FULL_DEPS_LIST})
```
Comment (KyleFromKitware) [inline tribits/core/package_arch/TribitsSetUpEnabledOnlyDependencies.cmake:173]: `IN LISTS`

File: tribits/core/package_arch/TribitsSetUpEnabledOnlyDependencies.cmake [hunk 6]
```diff
@@ -0,0 +1,205 @@
...
+
+    #print_var(${DEP_PACKAGE}_PKG_IDX)
+    set(DEP_PACKAGE_VALUE  ${${DEP_PACKAGE}_PKG_IDX})
+
+    set(SORTED_INDEX 0)
+    set(INSERTED_DEP_PACKAGE FALSE)
+
+    foreach(SORTED_PACKAGE  ${ORDERED_PACKAGE_FULL_DEPS_LIST})
```
Comment (KyleFromKitware) [inline tribits/core/package_arch/TribitsSetUpEnabledOnlyDependencies.cmake:181]: `IN LISTS`

**Review by KyleFromKitware — CHANGES_REQUESTED** [review]
Looks good, requested a few changes.

**Comment by bartlettroscoe** [comment]
Hello @KyleFromKitware, all of the issues you raised in your review [above]() should be addressed in the new commit https://github.com/TriBITSPub/TriBITS/pull/531/commits/35d2b978ab2e0b4b43fd5b52f777317f3f1ba4d3.  Can you please re-review and approve if issues are addressed?

**Pull request merged**